activation/poetdb.go

```diff
@@ -51,27 +52,54 @@ func (db *PoetDb) ValidateAndStorePoetProof(proof types.PoetProof, poetId [types
 		return true, fmt.Errorf("failed to store poet proof for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
-	if err := batch.Put(makeKey(poetId, roundId), ref[:]); err != nil {
+	key := makeKey(poetId, roundId)
+	if err := batch.Put(key[:], ref[:]); err != nil {
 		return true, fmt.Errorf("failed to store poet proof index entry for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
 	if err := batch.Write(); err != nil {
 		return true, fmt.Errorf("failed to store poet proof and index for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
-
+	db.log.Info("stored proof for round %d PoET id %x", roundId, poetId)
+	db.publishPoetProofRef(key, ref[:])
 	return false, nil
 }
 
-func (db *PoetDb) GetPoetProofRef(poetId [types.PoetIdLength]byte, roundId uint64) ([]byte, error) {
-	poetRef, err := db.store.Get(makeKey(poetId, roundId))
+func (db *PoetDb) SubscribeToPoetProofRef(poetId [types.PoetIdLength]byte, roundId uint64) chan []byte {
+	ch := make(chan []byte)
+	key := makeKey(poetId, roundId)
+	db.poetProofRefSubscriptions[key] = append(db.poetProofRefSubscriptions[key], ch)
+
+	go func() {
+		poetProofRef, err := db.getPoetProofRef(key)
+		if err != nil {
+			return
+		}
+		db.publishPoetProofRef(key, poetProofRef)
```

Comment:
there is a race here. It is possible that publish is called twice without protecting the map.